src/os/tempfile.go

```diff
@@ -1,5 +1,94 @@
 package os
 
+import (
+	"errors"
+	"internal/itoa"
+	"time"
+)
+
+// export for testing
+var ErrPatternHasSeparator = errPatternHasSeparator
```

Comment:
Things that are exported for testing only (and not otherwise as part of the public API) are normally put in `export_test.go`.  That file will get included when testing, and ignored otherwise.

Reply:
TIL!

Reply:
Alas, tinygo doesn't seem to support export_test.go.

Reply:
Hmm.. pretty sure I tested this before commenting. Let me double check.

Reply:
WorksForMe(tm)

```
diff --git a/src/os/tempfile.go b/src/os/tempfile.go
index b5452248..c88ec04a 100644
--- a/src/os/tempfile.go
+++ b/src/os/tempfile.go
@@ -6,9 +6,6 @@ import (
        "time"
 )

-// export for testing
-var ErrPatternHasSeparator = errPatternHasSeparator
-
 // fake_fastrand() is a placeholder for fastrand()
 // TODO: provide fastrand in runtime, as go does.  It is hard to implement properly elsewhere.
 // See https://github.com/tinygo-org/tinygo/issues/1037
~/go/src/github.com/tinygo-org/tinygo/src/os $ cat export_test.go
package os

// export for testing
var ErrPatternHasSeparator = errPatternHasSeparator
~/go/src/github.com/tinygo-org/tinygo/src/os $ ../../tinygo clean && ../../tinygo test os
ok  	os	0.373s

```

Reply:
OMG I have been doing it wrong for months. 

I  have been informed that the way to test code in the src/os directory is not, in fact, to run "tinygo test" in the src/os directory as one might expect, but instead to run "tinygo test os" anywhere in the source tree.

Running "tinygo test" in the src/os directory somewhat amazingly runs the tests on github.com/tinygo-org/tinygo/src/os rather than on the local files, hence my confusion.